### ISSUE TITLE
How can I restrict CardSettingsDouble decimal part to allow only 2 or 4 digits?

### ISSUE THREAD

**Issue by NTMS2017** [codegrue/card_settings#81]
How can I restrict CardSettingsDouble decimal part to allow only 2 or 4 digits?

**Comment by NTMS2017** [comment]
In class CardSettingsDouble extends CardSettingsText  there is a inputFormatters part. I modified this part and I add a DecimalTextInputFormatter class and also I import dart.math. I can add decimalRange properties so user can set the decimal places length. Currently in my project I only use 2. So I am hoping next version you can update this part for all of us.

Now when I type 12.987 it doesn't allow me to enter more than 2 decimal places, which perfectly okay with me now.


```
// import dart:math
import 'dart:math' as math;

CardSettingsDouble extends CardSettingsText {
...
...
...
       inputFormatters: [
            LengthLimitingTextInputFormatter(maxLength),
            WhitelistingTextInputFormatter(RegExp("[0-9]+.?[0-9]*")),
            //add DecimalTextInputFormatter
            DecimalTextInputFormatter(decimalRange: 2)
          ],
        );
}


//add DecimalTextInputFormatter class
class DecimalTextInputFormatter extends TextInputFormatter {
  DecimalTextInputFormatter({this.decimalRange})
      : assert(decimalRange == null || decimalRange > 0);

  final int decimalRange;

  @override
  TextEditingValue formatEditUpdate(
      TextEditingValue oldValue, // unused.
      TextEditingValue newValue,
      ) {
    TextSelection newSelection = newValue.selection;
    String truncated = newValue.text;

    if (decimalRange != null) {
      String value = newValue.text;

      if (value.contains(".") &&
          value.substring(value.indexOf(".") + 1).length > decimalRange) {
        truncated = oldValue.text;
        newSelection = oldValue.selection;
      } else if (value == ".") {
        truncated = "0.";

        newSelection = newValue.selection.copyWith(
          baseOffset: math.min(truncated.length, truncated.length + 1),
          extentOffset: math.min(truncated.length, truncated.length + 1),
        );
      }

      return TextEditingValue(
        text: truncated,
        selection: newSelection,
        composing: TextRange.empty,
      );
    }
    return newValue;
  }
}
```

**Comment by codegrue** [comment]
I like this. Would you be able to do a pull request? Is the whitelist formatter needed now that the decimal formatter is there?
________________________________

**Comment by NTMS2017** [comment]
Hi, I can make pull request having problem with company firewall. I am not sure if I understand your second question but I try to explain.

If add the '**class DecimalTextInputFormatter extends TextInputFormatter**' class in CardSettingsDouble you can add decimal length properties. So instead of hard code like this:

`DecimalTextInputFormatter(decimalRange: 2)`

we can write like

`DecimalTextInputFormatter(decimalRange: maxLenOfDecimal)`


Because of in my project all the CardSettingsDouble decimal point must be 2. So this hard code works for me. But if anyone needs more decimal places like 4 (example for currency converter) it will be nice to have this properties to be added.

So we can be able to write like this:

 ```
 CardSettingsDouble _buildCardSettingsDoubleTransferAmount() {
    return CardSettingsDouble(
        showMaterialIOS: true,
        unitLabel: 'USD',
        maxLenOfDecimal: 2
       ...
       ...

```

**Comment by codegrue** [comment]
I will add this manually.

**Comment by NTMS2017** [comment]
Thanks

**Comment by codegrue** [comment]
fixed in 1.5.0

**Comment by NTMS2017** [comment]
Thanks, I was reading the changeling this morning :) Kind Regards
